python-ecosys/urequests/urequests.py

```diff
@@ -130,3 +190,15 @@ def patch(url, **kw):
 
 def delete(url, **kw):
     return request("DELETE", url, **kw)
+
+
+def encode_basic_auth(username, password):
+    import ubinascii
+
+    formated = b"{}:{}".format(username, password)
+    formated = ubinascii.b2a_base64(formated)[:-1].decode("ascii")
+    return {"Authorization": "Basic {}".format(formated)}
+
+
+def is_chunked_data(data):
+    return getattr(data, "__iter__", None) and not getattr(data, "__len__", None)
```

Comment:
Does this need to be a separate function?  Smaller code if it's not.

Reply:
inlined now